fix(feeds): RSS improve torrent size parsing

ghoto opened the pull request:
When some RSS feeds have `Enclosure->Length` fixed size, autobrr fails to retrieve the size that's included in `Custom->Size`.

The previous logic used `custom->size` when `enclosure->length` was 0. But as stated in a previous issue, this `enclosure->length` can be a fixed length unrelated to the torrent. Therefore this value can be 0, fixed length, or the real one.

This modification will pick `custom->size` whenever is defined and it's bigger than `enclosures->length`.

https://github.com/autobrr/autobrr/issues/636

KyleSanderson commented on internal/feed/rss.go:
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/autobrr/autobrr/internal/release"
 	"github.com/autobrr/autobrr/pkg/errors"
 
+	"github.com/dustin/go-humanize"
```
Can you revert this?

KyleSanderson commented on internal/feed/rss.go:
```diff
@@ -154,10 +155,12 @@ func (j *RSSJob) processItem(item *gofeed.Item) *domain.Release {
+	
+	// When custom->size and enclosures->size differ, pick the largest one.
+	if size, ok := item.Custom["size"]; ok {
+		s, err := humanize.ParseBytes(size)
```
Ditto.

KyleSanderson reviewed (changes requested):
Thank you very much for the contribution. Can you please revert the module changes (if there's a problem with the existing function, that should be fixed instead).

ghoto commented:
What do you mean with the existing function?

ghoto commented:
@KyleSanderson I moved the logic to `ParseSizeBytesString`. I fixed another potential bug on the function that would reset the size of the Release to 0 if the string passed to the function was malformed.

zze0s reviewed (commented):
If we are making changes then might be good to check for not only enclosure size difference but quite a bit of feeds set the enclosure size to a static `39399` on all items.

While it could happen that something actually have that exact size, it's imo unlikely.
